src/main/java/com/segment/analytics/android/integrations/firebase/FirebaseIntegration.java

```diff
@@ -3,88 +3,278 @@
+    propertyMapper.put("order_id", Param.TRANSACTION_ID);
+    propertyMapper.put("currency", Param.CURRENCY);
+    return propertyMapper;
+  }
+
+  public FirebaseIntegration(Context context, Logger logger) {
+    this.logger = logger;
+    firebaseAnalytics = FirebaseAnalytics.getInstance(context);
+  }
+
+  @Override
+  public void onActivityResumed(Activity activity) {
+    super.onActivityResumed(activity);
 
-  private static final String FIREBASE_ANALYTICS_KEY = "Firebase Analytics";
+    PackageManager packageManager = activity.getPackageManager();
+    try {
+      ActivityInfo info =
+              packageManager.getActivityInfo(activity.getComponentName(), PackageManager.GET_META_DATA);
+      CharSequence activityLabel = info.loadLabel(packageManager);
+      /** setCurrentScreen should only be called in the onResume() activity */
+      firebaseAnalytics.setCurrentScreen(
+              activity, activityLabel.toString(), null /* class override */);
+      logger.verbose(
+              "firebaseAnalytics.setCurrentScreen(%s, %s, null /* class override */);",
+              activity, activityLabel.toString());
 
-  public FirebaseIntegration(Context context) {
-    mFirebaseAnalytics = FirebaseAnalytics.getInstance(context);
+    } catch (PackageManager.NameNotFoundException e) {
+      throw new AssertionError("Activity Not Found: " + e.toString());
+    }
   }
 
-  @Override public void onActivityStarted(Activity activity) {
-    super.onActivityStarted(activity);
+  @Override
+  public void identify(IdentifyPayload identify) {
+    super.identify(identify);
 
+    if (!isNullOrEmpty(identify.userId())) {
+      firebaseAnalytics.setUserId(identify.userId());
+      Map<String, Object> traits = identify.traits();
+      for (Map.Entry<String, Object> entry : traits.entrySet()) {
+        String trait = entry.getKey();
+        trait = trait.trim().replaceAll(" ", "_");
+        if (trait.length() > 40) {
+          trait = trimKey(trait);
+        }
+        String formattedValue;
+        if (entry.getValue() instanceof Date) {
+          Date value = (Date) entry.getValue();
+          formattedValue = formatDate(value);
+        } else if (entry.getValue() instanceof String && ISO_DATE.matcher(String.valueOf(entry.getValue())).matches()) {
+          formattedValue = String.valueOf(entry.getValue()).substring(0, 10);
+        } else {
+          formattedValue = String.valueOf(entry.getValue());
+        }
+
+        firebaseAnalytics.setUserProperty(trait, formattedValue);
+        logger.verbose("firebaseAnalytics.setUserProperty(%s, %s);", trait, formattedValue);
+      }
+    }
   }
 
-  @Override public void onActivityStopped(Activity activity) {
-    super.onActivityStopped(activity);
+  @Override
+  public void screen(ScreenPayload screen) {
+    super.screen(screen);
 
+    logger.verbose(
+            "The Firebase SDK gathers screen calls automatically. Segment does map manual screen " +
```

Comment:
I think @ladanazita already pointed out this isn't needed. I agree.

If we are going to keep this, the text should be updated. Right now it says "Segment does map manual screen methods to Firebase" - but it doesn't.